swri_rospy/src/swri_rospy/__init__.py

```diff
@@ -62,3 +63,32 @@ def wrapper(*args, **kwds):
             rospy.logerr(traceback.format_exc())
             return None
     return wrapper
+
+
+def subscribe_single_threaded(name, data_class, callback=None,
```

Comment:
Agreed that a class is ultimately better. I didn't at first because it seemed unnecessarily complicated in the YAGNI sense. Having now implemented one, it's no more complicated than the function, because subclassing `rospy.Subscriber` gets you all its methods for free.

Another option is a top-level class that owns a `rospy.Subscriber`, which would offer more flexibility, but require (for now) implementing pass-though methods. Fortunately, there are only two.

Both are easy; what do you think?